Add electron example app

tortmayr opened the pull request:
<!--
Thank you for your Pull Request. Please provide a description and review
the requirements below.

Contributors guide: https://github.com/eclipse-glsp/glsp/blob/master/CONTRIBUTING.md

Note: Security vulnerabilities should not be disclosed on GitHub, through a PR or any
other means. See [SECURITY.md](https://github.com/eclipse-glsp/glsp/blob/master/SECURITY.md),
to learn how to report vulnerabilities.
-->

#### What it does
- Add an electron example app in addition to the browser app. This allows use to also test the electron use case and makes it easier to discover electron specific issues.
- Adapt the build scripts to the same convention as used in the Theia repository.
   - yarn (install) only compiles the ts sources but does not build any example app
   - Example apps have to be built explicitly with `yarn browser build` or `yarn electron build`
- Rename glsp debug arugment to avoid clashes with nodes debug argument
- Update readmes and launch configs
- Update Jenkins file
<!-- Include relevant issues and describe how they are addressed. -->

#### How to test

<!-- Explain how a reviewer can reproduce a bug, test new functionality or verify performance improvements. -->
Build the  the electron app 
eg:
`yarn && yarn electron build`
Then test the different connection scenarions
`yarn electron start`
`yarn electron start:ws`
`yarn electron start:debug`
`yarn electron start:ws:debug`
`yarn electron start:integrated`

#### Follow-ups

<!-- Please list potential follow-up work, including known issues, possible future work, identified technical debt, and potentially introduced technical debt. If the PR introduces technical debt, specify the reason why this is acceptable. Please create tickets and link them here. Please use the label "technical debt" for new issues when it applies. -->

#### Changelog

<!-- Please check, when if it applies to your change. -->

-   [] This PR should be mentioned in the changelog
-   [ ] This PR introduces a breaking change (if yes, provide more details below for the changelog and the migration guide)

martin-fleck-at commented on package.json:
```diff
@@ -23,17 +23,9 @@
-    "start:ws": "yarn browser start:ws",
-    "start:ws:debug": "yarn browser start:ws:debug",
     "upgrade:next": "glsp updateNext",
     "watch": "concurrently --kill-others -n tsc,browser -c red,yellow \"tsc -b -w --preserveWatchOutput\" \"yarn -s browser watch\""
```
nit: The `watch` only goes to the browser, should that also work for electron? If so we should either split it into `watch:browser` and `watch:electron` or simply add a `watch:electron`.

tortmayr replied:
Its now splitted

martin-fleck-at reviewed (changes requested):
Change looks good to me and I can open the Electron app and switch between Electron and Browser without issues. Thank you Tobias! I only noticed two things which may or may not be fixed as part of this PR:

- The toolbox styling is customized for Theia but I'm not sure we want all of them. If we do it is fine and we definitely want to use Theia color variables etc. But could you please double check whether all the Theia tool palette styling is as it should be.

- The grid does not work in the Electron case. It does work for the browser but I believe something does work and this stands out: `stroke="color-mix(in srgb, %23000000 10%, transparent)"`. This might be "fixed" with https://github.com/eclipse-glsp/glsp-client/pull/364 and some minor adaptations though, what do you think?

tortmayr commented:
Thanks for the review @martin-fleck-at .
I have updated the PR and addressed the remaining issues.
- Toolbox styling has been adapted according to the latest client changes
- Grid styling now works for all theme types in browser and electron

I also sneaked in a minor improvement to the `createDiagramWidgetFactory` so that it can also be reused for configuring a widget factory for a subclass of GLSPDiagramWidget.

martin-fleck-at reviewed (approved):
Thank you for the quick update! I can confirm that the toolbox now looks nicer again and that the grid works in all themes. Great work, thank you, Tobias!
